skfp/model_selection/splitters/maxmin_split.py

```diff
@@ -0,0 +1,224 @@
+    ]
+
+    picker = MaxMinPicker()
+    test_idxs = picker.LazyPick(
+        distFunc=lambda i, j: 1
+        - DataStructs.DiceSimilarity(fingerprints[i], fingerprints[j]),
+        poolSize=data_size,
+        pickSize=test_size,
+        seed=random_state,
+    )
+
+    train_idxs = list(set(range(data_size)) - set(test_idxs))
+    train_subset: list[Any] = []
+    test_subset: list[Any] = []
+
+    if return_indices:
+        train_subset = train_idxs
+        test_subset = test_idxs
+    else:
+        train_subset = get_data_from_indices(data, train_idxs)
+        test_subset = get_data_from_indices(data, test_idxs)
+
+    ensure_nonempty_subset(train_subset, "train")
+    ensure_nonempty_subset(test_subset, "test")
+
+    if additional_data:
+        additional_data_split: list[Sequence[Any]] = split_additional_data(
+            list(additional_data), train_idxs, test_idxs
+        )
+        return train_subset, test_subset, additional_data_split
+    else:
+        return train_subset, test_subset
+
+
+@validate_params(
+    {
+        "data": ["array-like"],
+        "additional_data": ["tuple"],
+        "train_size": [
+            Interval(RealNotInt, 0, 1, closed="neither"),
+            Interval(Integral, 1, None, closed="left"),
+            None,
+        ],
+        "test_size": [
+            Interval(RealNotInt, 0, 1, closed="neither"),
+            Interval(Integral, 1, None, closed="left"),
+            None,
+        ],
+        "include_chirality": ["boolean"],
+        "return_indices": ["boolean"],
+    },
+    prefer_skip_nested_validation=True,
+)
+def maxmin_train_valid_test_split(
+    data: Sequence[Union[str, Mol]],
+    *additional_data: Sequence,
+    train_size: Optional[float] = None,
+    valid_size: Optional[float] = None,
+    test_size: Optional[float] = None,
+    return_indices: bool = False,
+    random_state: int = 0,
+) -> Union[
+    tuple[
+        Sequence[Union[str, Mol]], Sequence[Union[str, Mol]], Sequence[Sequence[Any]]
+    ],
+    tuple[Sequence, ...],
+    tuple[Sequence[int], Sequence[int]],
+]:
+
+    data_size = len(data)
+    train_size, valid_size, test_size = validate_train_valid_test_split_sizes(
+        train_size, valid_size, test_size, len(data)
+    )
+    molecules = ensure_mols(data)
+    fingerprints = [
+        AllChem.GetMorganFingerprintAsBitVect(mol, 2, 1024) for mol in molecules
+    ]
+    picker = MaxMinPicker()
+
+    test_idxs = picker.LazyPick(
+        distFunc=lambda i, j: 1
+        - DataStructs.DiceSimilarity(fingerprints[i], fingerprints[j]),
+        poolSize=data_size,
+        pickSize=test_size,
+        seed=random_state,
+    )
+
+    valid_idxs = picker.LazyPick(
+        distFunc=lambda i, j: 1
+        - DataStructs.DiceSimilarity(fingerprints[i], fingerprints[j]),
+        poolSize=data_size,
+        pickSize=test_size + valid_size,
+        firstPicks=test_idxs,
+        seed=random_state,
+    )
```

Comment:
Add a comment for `firstPicks` and how this works. This is smart and efficient, but not obvious

Reply:
done